x/config/util_for_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package config
+
+import "strings"
+
+// trims the leading tabs on every line of a block of text, and converts all
+// remaining tabs to two spaces, such that YAML accepts the whole block.
+func expand(s string) string {
+	lines := strings.Split(s, "\n")
+	var i, c int
+
+	// find first line's depth
+	for ; i < len(lines); i++ {
+		c = depth(lines[i])
+		if c != 0 {
+			break
+		}
+		lines[i] = ""
+	}
+
+	// trim and expand remaining lines
+	for ; i < len(lines); i++ {
+		d := depth(lines[i])
+		if c < d {
+			d = c
+		}
+		lines[i] = expandLine(lines[i][d:])
+	}
+
+	return strings.Join(lines, "\n")
+}
+
+func expandLine(l string) string {
+	r := ""
+	i := 0
+	for ; i < len(l); i++ {
+		switch l[i] {
+		case '\t':
+			r += "  "
+		default:
+			return r + l[i:]
```

Comment:
optional: this can probably be simplified to,

```
if i := strings.IndexFunc(s, notTab); i >= 0 {
  return strings.Repeat("  ", i) + l[i:]
}

// where notTab := func(r rune) bool { return r != '\t' }
```

And avoid the append to `r`.